### DISCUSSION TITLE
Content security policy nonce for styles

### DISCUSSION THREAD

**Discussion by duffn** [rappasoft/laravel-livewire-tables#592] [Feature Requests]
First, thanks for the great package.

I'm attempting to use the bootstrap4 theme (though this would happen with bootstrap5, too) along with a content security policy that allows `unsafe-inline` styles with `laravel-csp`: https://github.com/spatie/laravel-csp

This is not possible with the styles in the templates. https://github.com/search?q=style%3D+repo%3Arappasoft%2Flaravel-livewire-tables+extension%3Aphp&type=Code&ref=advsearch&l=&l=

Ideally these would all be in an external stylesheet, but one other potential solution is to convert these to `<style></style>` tags and then provide an option to set a nonce on the style tags like Livewire does here: https://github.com/livewire/livewire/blob/f2e2c341394ee4aec98a28373ca73ba1bf6be450/src/LivewireManager.php#L192-L195. I'm successfully using this functionality with Livewire and `laravel-csp`. 

If this is a change you'd want, I could take a crack at it.

**Comment by liepumartins** [comment]
I second this!

My current workaround for this:
1. Publish all views
`php artisan vendor:publish --provider="Rappasoft\LaravelLivewireTables\LaravelLivewireTablesServiceProvider" --tag=livewire-tables-views`
2. Search for `style=` and replace with appropriate CSS class. I had to create custom (tailwind like) classes like w-4, h-5 etc.
